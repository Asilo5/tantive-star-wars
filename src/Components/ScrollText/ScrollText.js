import React from 'react';
import './ScrollText.scss';
import { Link } from 'react-router-dom';

const ScrollText = ({ movieScroll }) => {
    return (
      <section>
          <section className='scroll-text'>
            <div className="fade"></div>
            <section className='star-wars-text'>
              <div className='crawl'>
              <div className='title-text'>
                <p>Episode {movieScroll.episode_id}</p>
                <h3>{movieScroll.title}</h3>
              </div>
              {movieScroll.openingCrawl}
            </div>
          </section>
      </section>
       <Link to='/movies/:id'>
         <button className="scroll-button" >Continue to see characters..</button>
       </Link>
      </section>
    )
}

export default ScrollText;