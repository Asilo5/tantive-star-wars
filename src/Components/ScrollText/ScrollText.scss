$accent_color: #00D800;

.scroll-text {
  overflow: hidden;
  background-repeat: no-repeat;
  margin: 58px 20px;
  border-radius: 10px;
}

.title-text {
  text-align: center;
  margin-bottom: 2em;
  p {
    margin-bottom: 10px;
  }
  h3 {
    font-size: 2em;
    margin: 10px;
  }
}

.star-wars-text {
  display: flex;
  height: 100px;
  perspective: 400px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 400%;
  font-weight: 600;
  letter-spacing: -4px;
  line-height: 160%;
  text-align: justify;
}

.fade {
    position: relative;
    width: 100%;
    min-height: 60vh;
    top: -2px;
    background-image: linear-gradient(0deg, transparent, black 75%);
    z-index: 1;
  }

.crawl {
  position: relative;
  top: -100px;
  animation: crawl 80s linear;
}

.scroll-button {
  border-radius: 3px;
  font-size: 2em;
  font-weight: 700;
  width: 35%;
  padding: 20px;
  color: $accent_color;
  border: 2px solid $accent_color;
  &:hover {
      font-weight: 800;
      cursor: pointer;
      transform: scale(1.1);
      transition: all .3s ease-in-out;
    }
}

@keyframes crawl {
    0% {
        top: 0;
        transform: rotateX(20deg) translateZ(0);
    }
    100% {
        top: -6000px;
        transform: rotateX(25deg) translateZ(-2500px);
    }
}

