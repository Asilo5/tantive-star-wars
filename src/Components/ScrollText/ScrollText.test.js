import React from 'react';
import { shallow } from 'enzyme';
import ScrollText from './ScrollText';

describe('ScrollText', () => {

    let wrapper;
    let mockMovieScroll = {
        id: 1,
        title: 'Star Wars',
        crawl: 'Learning curve business-to-business first mover advantage ecosystem scrum project niche market investor customer return on investment buzz business model canvas bootstrapping technology handshake.'
    }

    beforeEach(() => {
        wrapper = shallow(<ScrollText movieScroll={mockMovieScroll} />)
    })

    it('should match snapshot with all the information passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
})