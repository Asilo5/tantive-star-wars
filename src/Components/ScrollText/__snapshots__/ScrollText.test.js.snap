// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollText should match snapshot with all the information passing in correctly 1`] = `
<section>
  <section
    className="scroll-text"
  >
    <div
      className="fade"
    />
    <section
      className="star-wars-text"
    >
      <div
        className="crawl"
      >
        <div
          className="title-text"
        >
          <p>
            Episode 
          </p>
          <h3>
            Star Wars
          </h3>
        </div>
      </div>
    </section>
  </section>
  <Link
    to="/movies/:id"
  >
    <button
      className="scroll-button"
    >
      Continue to see characters..
    </button>
  </Link>
</section>
`;
